### PR TITLE
make user-agent and content-type overwritable with undefined

### PR DESCRIPTION
**Pull request by Uzlopak**
@budarin
I open this PR to see if the unit tests run.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by budarin** [comment]
Thanks!

This covers the necessary requirements 🚀

File: test/test.js [hunk 1]
```diff
@@ -1924,3 +1924,28 @@ test('should work with pipeline', (t) => {
     t.equal(res.rawPayload.toString(), 'example.com:8080|/hello')
   })
 })
+
+test('should leave the headers user-agent and content-type undefined when autoFillHeaders is set to false', (t) => {
```
Comment (Uzlopak) [inline test/test.js:1928]:
```suggestion
test('should leave the headers user-agent and content-type undefined when the headers are explicitly set to undefined in the inject', (t) => {
```

**Comment by Uzlopak** [comment]
Well it is based on your work. Lets see if the others agree

**Review by Eomm — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**